drivers/input/input_cf1133.c

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright (c) 2023, Kickmaker
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*Address by default is 0x55*/
+
+#define DT_DRV_COMPAT sitronix_cf1133
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/input/input.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(cf1133, CONFIG_INPUT_LOG_LEVEL);
+
+/* cf1133 used registers */
+typedef enum {
+	FIRMWARE_VERSION,
+	STATUS_REG,
+	DEVICE_CONTROL_REG,
+	TIMEOUT_TO_IDLE_REG,
+	XY_RESOLUTION_HIGH,
+	X_RESOLUTION_LOW,
+	Y_RESOLUTION_LOW,
+	DEVICE_CONTROL_REG2 = 0x09,
+	FIRMWARE_REVISION_3 = 0x0C,
+	FIRMWARE_REVISION_2,
+	FIRMWARE_REVISION_1,
+	FIRMWARE_REVISION_0,
+	FINGERS,
+	KEYS_REG,
+	XY0_COORD_H,
+	X0_COORD_L,
+	Y0_COORD_L,
+	I2C_PROTOCOL = 0x3E,
+	MAX_NUM_TOUCHES,
+	DATA_0_HIGH,
+	DATA_0_LOW,
+	MISC_CONTROL = 0xF1,
+	SMART_WAKE_UP_REG = 0xF2,
+	CHIP_ID = 0xF4,
+	PAGE_REG = 0xff,
+} RegisterOffset;
+
+/* Constants  */
+#define SITRONIX_MAX_SUPPORTED_POINT 10U
+
+/* Mask  */
+#define ONE_D_SENSING_CONTROL_SHFT 0x2
+#define ONE_D_SENSING_CONTROL_BMSK 0x3
+#define I2C_PROTOCOL_BMSK          0x3
+
+/* Useful enum  */
+typedef enum {
+	XY_COORD_H,
+	X_COORD_L,
+	Y_COORD_L,
+	PIXEL_DATA_LENGTH_B,
+	PIXEL_DATA_LENGTH_A,
+} PIXEL_DATA_FORMAT;
+
+typedef enum {
+	SITRONIX_RESERVED_TYPE_0,
+	SITRONIX_A_TYPE,
+	SITRONIX_B_TYPE,
+} I2C_PROTOCOL_TYPE;
+
+/** CF1133 configuration (DT). */
+struct cf1133_config {
+	/** I2C bus. */
+	struct i2c_dt_spec bus;
+	struct gpio_dt_spec reset_gpio;
+#ifdef CONFIG_INPUT_CF1133_INTERRUPT
+	/** Interrupt GPIO information. */
+	struct gpio_dt_spec int_gpio;
+#endif
+};
+
+/** CF1133 data. */
+struct cf1133_data {
+	/** Device pointer. */
+	const struct device *dev;
+	/** Work queue (for deferred read). */
+	struct k_work work;
+#ifdef CONFIG_INPUT_CF1133_INTERRUPT
+	/** Interrupt GPIO callback. */
+	struct gpio_callback int_gpio_cb;
+#else
+	/** Timer (polling mode). */
+	struct k_timer timer;
+#endif
+	/** Last pressed state. */
+	bool pressed_old;
+	bool pressed;
+
+	int resolution_x;
+	int resolution_y;
+	uint8_t max_touches;
+	uint8_t touch_protocol_type;
+	uint8_t pixel_length;
+	uint8_t chip_id;
+
+	uint8_t Num_X;
+	uint8_t Num_Y;
+	uint8_t sensing_mode;
+};
+
+static int cf1133_get_chip_id(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int ret = 0;
```

Comment:
```suggestion
	int ret;
```

no need to initialize this one

Reply:
Done